tempto-core/src/main/java/io/prestodb/tempto/internal/convention/AnnotatedFileParser.java

```diff
@@ -143,11 +144,22 @@ private int findNextSectionIndex(List<String> lines)
      */
     private List<String> filterContent(List<String> contentLines)
     {
-        contentLines = contentLines.stream()
-                .filter(s -> !(isSpecialLine(s) || isBlank(s)))
-                .map(AnnotatedFileParser::unescapeLine)
-                .collect(toList());
-        return contentLines;
+        ImmutableList.Builder<String> result = ImmutableList.builder();
+
+        for (int i = 0; i < contentLines.size(); i++) {
+            String line = contentLines.get(i);
+            if (isSpecialLine(line) || isBlank(line)) {
+                continue;
+            }
+
+            while (line.endsWith(LINE_ESCAPE)) {
+                checkState(i + 1 < contentLines.size(), "There no more lines, while the end of the line was escaped");
+                line = line.substring(0, line.length() - 1) + "\n" + contentLines.get(++i);
```

Comment:
- `line.substring(0, line.length() - 1)` is hardly readable. i would use `replaceAll(".$", "")` instead
- don't increment `i` inline (i.e. don't treat `i++` or `++i` as an expression)
- this should be a `while` loop -- you shouldn't modify `for`'s loop index variablae
- this is quadratic